### PR TITLE
feat(bundling): add convert-to-inferred generator for @nx/rollup

### PR DESCRIPTION
**Pull request by jaysoo**
This PR adds `@nx/rollup:convert-to-inferred` generator.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+
+import { extractRollupConfigFromExecutorOptions } from './lib/extract-rollup-config-from-executor-options';
+import { RollupExecutorOptions } from '../../executors/rollup/schema';
+
+interface Schema {
+  project?: string;
+  all?: boolean;
```
Comment (Coly010) [inline packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts:14]: We can remove this, it should be the default behaviour of the generator. Rather than forcing a user to potentially pass it. We can update the conditional below to only check if project specifically is called.

File: packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  let migrated = 0;
+
+  const projects = getProjects(tree);
+  const originalDefaultOptions = new Map<string, RollupExecutorOptions>();
+
+  forEachExecutorOptions<RollupExecutorOptions>(
+    tree,
+    '@nx/rollup:rollup',
+    (currentTargetOptions, projectName, targetName, configurationName) => {
+      if (!options.all && options.project && projectName !== options.project)
```
Comment (Coly010) [inline packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts:28]: As above, we can shorten this to `options.project && projectName !== options.project`.

A user specifying `--project` should be the only flag to determine if all projects are migrated, or if a specific project is migrated

File: packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+      const project = projects.get(projectName);
+      const target = project.targets[targetName];
+      console.log(
+        projectName,
+        targetName,
+        configurationName,
+        JSON.stringify(project, null, 2)
+      );
+
```
Comment (Coly010) [inline packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts:39]: We should remove this in the final version

**Review by Coly010 — APPROVED** [review]
One change, but it's minor

**Review by ndcunningham — APPROVED** [review]
